conbench/tests/api/test_compare.py

```diff
@@ -315,13 +315,38 @@ def test_compare_unknown_compare_ids(self, client):
         response = client.get("/api/compare/benchmarks/foo...bar/")
         self.assert_404_not_found(response)
 
+    @pytest.mark.parametrize(
+        ["baseline_result_id", "expected_z_score"],
+        [
+            # result on the fork point commit
+            (5, -2.1857497724635935),
+            # result on the parent commit
+            (6, -2.8246140882627757),
+            # result on the head commit of the default branch
+            (8, 0.12045310863100521),
+        ],
+    )
+    def test_compare_different_baselines(
+        self, client, baseline_result_id, expected_z_score
+    ):
+        self.authenticate(client)
+        _, benchmark_results = _fixtures.gen_fake_data()
+        contender = benchmark_results[7]
```

Comment:
Is the contender on a non-default branch?

I think it would be valuable if you could clarify the relationship of this test with #992! Is #992 resolved now as a side effect of this patch?

Reply:
#992 is not resolved, and it becomes much more important to resolve as a result of this patch! Because now, workflows like `benchalerts` that do the following might break even worse:
1. get a contender run id
2. hit `/api/runs/{contender_run_id}` to get its baseline run id
3. hit `/api/compare/runs/{baseline_run_id}...{contender_run_id}` to get comparisons

#992 means that for contender runs on a PR branch with multiple runs on multiple commits, step 2 could return another PR run on the parent commit as the baseline run. Before this patch, it didn't matter, because step 3 always returned the intended comparisons (with the baseline commit being the fork-from-default). But now, step 3 will return a comparison that includes the PR runs in the baseline distribution. That could pollute the z-scores.